src/schnorrsig.rs

```diff
@@ -449,6 +450,38 @@ impl<'de> ::serde::Deserialize<'de> for PublicKey {
     }
 }
 
+impl SecretKey {
+    /// Creates a new secret key using data from BIP-340 [`KeyPair`]
+    pub fn from_keypair<V: Verification>(secp: &Secp256k1<V>, keypair: &KeyPair) -> Self {
+        let mut sk = [0; constants::SECRET_KEY_SIZE];
+        unsafe {
+            let ret = ffi::secp256k1_keypair_sec(
+                secp.ctx,
+                sk.as_mut_c_ptr(),
+                keypair.as_ptr()
+            );
+            debug_assert_eq!(ret, 1);
+        }
+        SecretKey(sk)
+    }
+}
+
+impl ::key::PublicKey {
+    /// Creates a new compressed public key key using data from BIP-340 [`KeyPair`]
+    pub fn from_keypair<C: Signing>(secp: &Secp256k1<C>, keypair: &KeyPair) -> Self {
```

Comment:
Same here, `secp256k1_keypair_pub` doesn't require the context to be initialized for Signing, you can use the no precomp here too

Reply:
Fixed as above